user/advanced-configuration/managing-vm-kernel.md

```diff
@@ -302,25 +303,31 @@ sudo update-grub
 ~~~
 
 You can safely ignore this error message:
-
-~~~
-grub2-probe: error: cannot find a GRUB drive for /dev/mapper/dmroot. Check your device.map
-~~~
+`grub2-probe: error: cannot find a GRUB drive for /dev/mapper/dmroot. Check your device.map`
 
 You may want to adjust some settings in `/etc/default/grub` (or better `/etc/default/grub.d`). For example, lower `GRUB_TIMEOUT` to speed up VM startup. You need to re-run `sudo update-grub` after making grub confugration changes.
 
 Then shutdown the VM.
 
-Go to Qubes VM Manger -> right click on the VM -> Qube settings -> Advanced -> choose `pvgrub2-phv` -> OK
+Go to dom0 -> Qubes VM Manger -> right click on the VM -> Qube settings -> Advanced
+
+Depends on `Virtualization` mode setting:
+
+* `Virtualization` mode `PV`: Use of `Virtualization` mode `PV` mode is discouraged for security purposes.
+  * If you require `Virtualization` mode `PV` mode, install `grub2-xen` in dom0. This can be done by running command `sudo qubes-dom0-update grub2-xen` in dom0.
+* `Virtualization` mode `PVH`: Booting to a kernel inside a TemplateVM is unsupported.
```

Comment:
No, it is supported thanks to grub 2.04, packaged in `pvgrub2-pvh` package.

Reply:
how is it possible to install the pvgrub2-pvh package? because with qubes-dom0-update pvgrub2-pvh i only get nothing to download.

i only have the default linux 4.19 kernel
none 
pvgrub2

no pvgrub2-pvh so i cant virtualize pvh vms with managed kernel right now